lib/ykpiv.c

```diff
@@ -202,6 +203,53 @@ ykpiv_rc ykpiv_connect(ykpiv_state *state, const char *wanted) {
   return YKPIV_GENERIC_ERROR;
 }
 
+ykpiv_rc ykpiv_reconnect(ykpiv_state *state) {
+  unsigned long active_protocol;
+  long rc;
+
+  if(state->verbose) {
+    fprintf(stderr, "trying to reconnect to current reader.\n");
+  }
+  rc = SCardReconnect(state->card, SCARD_SHARE_SHARED,
+    SCARD_PROTOCOL_T1, SCARD_RESET_CARD, &active_protocol);
+  if(rc != SCARD_S_SUCCESS) {
+    if(state->verbose) {
+      fprintf(stderr, "SCardReconnect failed, rc=%08lx\n", rc);
+    }
+    return YKPIV_PCSC_ERROR;
+  }
+
+  {
```

Comment:
Might want to consider if the select applet part should be abstracted out to a common function called by this and ykpiv_connect().